sigstore/sign.py

```diff
@@ -296,7 +307,12 @@ class SigningContext:
     """
 
     def __init__(
-        self, *, fulcio: FulcioClient, rekor: RekorClient, trusted_root: TrustedRoot
+        self,
+        *,
+        fulcio: FulcioClient,
+        rekor: RekorClient,
+        trusted_root: TrustedRoot,
+        tsa_clients: List[TimestampAuthorityClient] | None = None,
```

Comment:
Flagging: it's not ideal that this parameter list continues to grow; I think we could probably slice it down substantially by passing the entire `ClientTrustConfig` and doing the instantiations internally.

That'd be good for a follow-on refactor PR here.